test_strip_protocol_expanduser assumes home paths

`test_strip_protocol_expanduser` assumes that `$USER` is part of the path of `$HOME`, but there's no requirement for that to be so.

Indeed, and that’s the only failing test for me:
```
________________________ test_strip_protocol_expanduser ________________________

    def test_strip_protocol_expanduser():
        path = "file://~\\foo\\bar" if sys.platform == "win32" else "file://~/foo/bar"
        stripped = LocalFileSystem._strip_protocol(path)
        assert path != stripped
        assert "file://" not in stripped
>       assert getpass.getuser() in stripped
E       AssertionError: assert 'builduser' in '/build/foo/bar'
E        +  where 'builduser' = <function getuser at 0x7f09ba45b670>()
E        +    where <function getuser at 0x7f09ba45b670> = getpass.getuser

fsspec/implementations/tests/test_local.py:472: AssertionError
```

--- a/fsspec/implementations/tests/test_local.py
+++ b/fsspec/implementations/tests/test_local.py
@@ -8,7 +8,6 @@ import sys
 from contextlib import contextmanager
 import posixpath
 import tempfile
-import getpass
 
 import pytest
 import fsspec
@@ -508,7 +507,7 @@ def test_strip_protocol_expanduser():
     stripped = LocalFileSystem._strip_protocol(path)
     assert path != stripped
     assert "file://" not in stripped
-    assert getpass.getuser() in stripped
+    assert stripped.startswith(os.path.expanduser("~"))
 
 
 def test_iterable(tmpdir):

